### PR TITLE
Let bricked DTPF accept primitive blast furnace recipes, use screwdri…

### PR DESCRIPTION
**Pull request by koiNoCirculation**
…ver to switch mode.

### CODE REVIEW

**Comment by Nxer** [comment]
@HoleFish

**Comment by Nxer** [comment]
How do the new normal primitive blast furnace do? Paralleled ? or something

**Review by HoleFish — CHANGES_REQUESTED** [review]

File: src/main/java/com/Nxer/TwistSpaceTechnology/common/machine/GT_TileEntity_MegaBrickedBlastFurnace.java [hunk 1]
```diff
@@ -578,6 +621,121 @@ public IStructureDefinition<GT_TileEntity_MegaBrickedBlastFurnace> getStructureD
...
+                if (!fuels.contains(itemWithDamage)) {
+                    result.originalRatio.put(itemWithDamage, ingredient.stackSize / (double) recipefuelAmount);
+                }
+            }
+        }
+
+        for (ItemStack itemStack : inputList) {
+            ItemWithDamage itemWithDamage = new ItemWithDamage(itemStack.getItem(), itemStack.getItemDamage());
+            int itemCount = itemCountInput.getOrDefault(itemWithDamage, 0) + itemStack.stackSize;
+            itemCountInput.put(itemWithDamage, itemCount);
```
[the remark this answers is not in the document]
Comment (HoleFish) [reply src/main/java/com/Nxer/TwistSpaceTechnology/common/machine/GT_TileEntity_MegaBrickedBlastFurnace.java:671]: `HashMap` does

File: src/main/java/com/Nxer/TwistSpaceTechnology/common/machine/GT_TileEntity_MegaBrickedBlastFurnace.java [hunk 2]
```diff
@@ -652,23 +658,23 @@ static MaterialConsumption calculateMaterialConsumption(GT_Recipe recipe, List<I
...
                     result.originalRatio.put(itemWithDamage, ingredient.stackSize / (double) recipefuelAmount);
                 }
             }
         }
 
         for (ItemStack itemStack : inputList) {
-            ItemWithDamage itemWithDamage = new ItemWithDamage(itemStack.getItem(), itemStack.getItemDamage());
+            TST_ItemID itemWithDamage = TST_ItemID.create(itemStack);
             int itemCount = itemCountInput.getOrDefault(itemWithDamage, 0) + itemStack.stackSize;
             itemCountInput.put(itemWithDamage, itemCount);
```
Comment (Nxer) [inline src/main/java/com/Nxer/TwistSpaceTechnology/common/machine/GT_TileEntity_MegaBrickedBlastFurnace.java:671]: Turn to use `merge` , seems the final.

**Review by Nxer — CHANGES_REQUESTED** [review]

**Review by Nxer — APPROVED** [review]

**Pull request merged**